language/proto/fileinfo.go

```diff
@@ -77,6 +81,26 @@ func protoFileInfo(dir, name string) FileInfo {
 		case match[serviceSubexpIndex] != nil:
 			info.HasServices = true
 
+			// match is of the format "service ServiceName {".
+			// extract just the service name
+			fullMatch := string(match[serviceSubexpIndex])
+			serviceName := strings.Fields(fullMatch)[1]
```

Comment:
This will `panic` if (for some reason) there's no space. Also, is the trailing space required?

I think this is valid:
```
service ServiceName{
```

Because of. this complexity, I think it would make sense to add this `extractObjectName()` to a function:

The function could then extract based on the situation:
```
func extractObject(fullMatch) {
  fields := fullMatch.Fields
  if len(fields) < 2 {
    // handle the error
  }
  return strings.TrimSuffix(fields[1], "{"
}
```

Reply:
That's a good call. The way the service regex is currently written, the space on either side is optional. This means that `service ABC{` would get picked up as a service, but also `message serviceABC {`  would also get picked up (tested this out and it did).

I think we should update all the regexes to require a space between service/message/enum and then trim the suffix of any potential leftover "{" like in your comment

Reply:
Updated in newest revision